.github/workflows/release.yaml

```diff
@@ -0,0 +1,39 @@
+name: Release to latest/edge
+
+on:
+  push:
+    branches:
+      - main
+    paths-ignore:
+      - ".github/renovate*"
+      - ".github/workflows/release.yaml"
+      - ".github/workflows/renovate.yaml"
+      - ".github/workflows/update-libs.yaml"
+      - ".gitignore"
+      - "tox.ini"
+
+jobs:
+  test:
+    uses: ./.github/workflows/build-and-test.yaml
+    secrets:
+      CHARMHUB_TOKEN: "${{ secrets.CHARMHUB_TOKEN }}"
+
+  release-to-charmhub:
+    name: Release to CharmHub
+    needs:
+      - test
+    runs-on: ubuntu-22.04
+    steps:
+      - name: Checkout
+        uses: actions/checkout@b4ffde65f46336ab88eb53be808477a3936bae11 # v4
+        with:
+          fetch-depth: 0
+      - name: Select charmhub channel
+        uses: canonical/charming-actions/channel@631c2d944da2bd12430f1f3a954c8fffcf2385cd # 2.4.0
+        id: channel
+      - name: Upload charm to charmhub
+        uses: canonical/charming-actions/upload-charm@631c2d944da2bd12430f1f3a954c8fffcf2385cd # 2.4.0
+        with:
+          credentials: "${{ secrets.CHARMHUB_TOKEN }}"
+          github-token: "${{ secrets.GITHUB_TOKEN }}"
+          channel: "${{ steps.channel.outputs.name }}"
```

Comment:
As you defined the `charmcraft-channel` in the release-charm action, would you do that when invoking the upload-charm too?

Reply:
The upload charm sets the channel to latest/edge if the branch under execution is the default branch for the repo

Reply:
The `charmcraft-channel` I mentioned above is the [snap channel to use when installing charmcraft](https://github.com/canonical/charming-actions/tree/main/upload-charm) not the channel the charm is going to be uploaded to